pkg/webhooks/workload_webhook.go

```diff
@@ -182,6 +184,87 @@ func validateContainer(c *corev1.Container, path *field.Path) field.ErrorList {
 	return allErrs
 }
 
+func validateAdmissionChecks(obj *kueue.Workload, basePath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	for i := range obj.Status.AdmissionChecks {
+		admissionChecksPath := basePath.Index(i)
+		ac := &obj.Status.AdmissionChecks[i]
+		allErrs = append(allErrs, validatePodSetUpdates(ac, obj, admissionChecksPath.Child("podSetUpdates"))...)
+	}
+	return allErrs
+}
+
+func validatePodSetUpdates(acs *kueue.AdmissionCheckState, obj *kueue.Workload, basePath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	knowPodSets := make(map[string]struct{}, len(obj.Spec.PodSets))
+	knowPodSetNames := make([]string, len(obj.Spec.PodSets))
+	for i := range obj.Spec.PodSets {
+		name := obj.Spec.PodSets[i].Name
+		knowPodSets[name] = struct{}{}
+		knowPodSetNames = append(knowPodSetNames, name)
+	}
+	for i := range acs.PodSetUpdates {
+		psu := &acs.PodSetUpdates[i]
+		psuPath := basePath.Index(i)
+		for _, msg := range validation.IsDNS1123Label(psu.Name) {
+			allErrs = append(allErrs, field.Invalid(psuPath.Child("name"), psu.Name, msg))
+		}
+		if _, found := knowPodSets[psu.Name]; !found {
```

Comment:
I think currently the implementation only works then the set in podSets equals the set in podSetUpdates. Not sure this is expected. @stuton please confirm.

Reply:
when the names of podSetUpdates don't equal the names of podSets we will provide "notsupported" error

Reply:
IIUC currently in code you only check that every name in podSetUpdates is in known set. So you are not checking equality of the sets, but you are checking that the set of names is podSetUpdates is a subset of the names in podSets.